### ISSUE TITLE
Daily reset softlock

### ISSUE THREAD

**Issue by SapiensAnatis** [SapiensAnatis/Dawnshard#55]
**Describe the bug**
If a device is left to idle overnight, and opened after 'reset', a message will appear stating 'a new day has dawned' and kick the player to the home screen (not the title screen). This is despite the server not giving any indication of a daily reset and not even having any concept of one -- perhaps it's based on the client's stored last login time.

Upon returning to the home screen, the user is UI locked with the following screen, softlocked with a glitched endeavour list

![Screenshot_20221225-181642_Dragalia](https://user-images.githubusercontent.com/5047192/209478548-225c0320-48b5-4cfb-8369-da9e0273cf06.jpg)

**On branch**
develop / deployed server 25/12/2022

**To Reproduce**
1. Log into the server
2. Leave the device idling past reset time in your locale
3. Get kicked to title screen

**Expected behavior**
At the very least it shouldn't softlock, but it looks like it's expecting an item in an UpdateDataList somewhere, so maybe we need to fix this by implementing login bonuses.

**Additional context**
ADB logcat:

```
12-25 18:16:39.943 24600 24645 E Unity   : NullReferenceException: Object reference not set to an instance of an object.
12-25 18:16:39.943 24600 24645 E Unity   :   at Gluon.DataManager.UpdateByUpdateDataList (Gluon.Http.UpdateDataList updateDataList) [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at Gluon.DataManager.UpdateByResponse (Gluon.Http.MypageInfoResponse res) [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at Gluon.MyPageScene.OnMyPageInfoSuccess (Gluon.Http.MypageInfoResponse response) [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at System.Action`1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at System.Action`1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at Gluon.Http.MypageInfoTask.Deserialize (System.Byte[] body) [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at Cute.Http.HttpManager.Cute.Core.IUpdatable.FastUpdate () [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :   at Cute.Core.UpdateDispatcher.Update () [0x00000] in <00000000000000000000000000000000>:0
12-25 18:16:39.943 24600 24645 E Unity   :
12-25 18:16:39.943 24600 24645 E Unity   : (Filename: currently not available on il
```

**Comment by SapiensAnatis** [comment]
To reproduce on demand: Return ResultCode.CommonChangeDate

**Comment by SapiensAnatis** [comment]
Closed by #282
